metrics-exporter-prometheus/src/builder.rs

```diff
@@ -361,6 +363,17 @@ impl PrometheusBuilder {
         self
     }
 
+    /// Sets the purge timeout for metrics.
+    ///
+    /// If a purge timeout is set, the purger will call `.render()` on the registry, causing
+    /// the values from histograms to be drained out. This ensures that stale histogram values
+    /// do not persist indefinitely.
```

Comment:
Similar note here about just changing the wording overall from "purge"/"purging" to "upkeep".

I would also remove the specific bit about calling `.render()`, since it doesn't actually do that anymore. Just be generic, something like:

```rust
/// Sets the upkeep interval.
///
/// The upkeep task handles periodic maintenance operations, such as draining histogram data,
/// to ensure that all recorded data is up-to-date and prevent unbounded memory growth. 
```